### ISSUE TITLE
Some objects are created with 'STANDARD' storage class instead of 'INTELLIGENT_TIERING' on s3 disk.

### ISSUE THREAD

**Issue by 1283386** [ClickHouse/ClickHouse#68551]
**Describe the unexpected behaviour**
Some objects are created with 'STANDARD' storage class instead of 'INTELLIGENT_TIERING' on s3 disk despite that s3_storage_class is set to INTELLIGENT_TIERING

**How to reproduce**
* Which ClickHouse server version to use
24.3.7.30 

* Non-default settings, if any
```
<clickhouse>
    <storage_configuration>
        <disks>
            <s3>
                <type>s3</type>
                <endpoint>https://clickhouse-offload.s3.amazonaws.com/{cluster}/{replica}/</endpoint>
                <s3_storage_class>INTELLIGENT_TIERING</s3_storage_class>
            </s3>
        </disks>
        <policies>
            <s3_policy>
                <volumes>
                    <s3_vol>
                        <disk>s3</disk>
                    </s3_vol>
                </volumes>
            </s3_policy>
        </policies>
    </storage_configuration>
</clickhouse>
```
* `CREATE TABLE` statements for all tables involved
```
CREATE TABLE default.test_s3_tiered
(
    `A` Int64,
    `S` String,
    `D` Date
)
ENGINE = MergeTree
PARTITION BY D
ORDER BY A
SETTINGS storage_policy = 's3_policy', index_granularity = 8192
```

* Queries to run that lead to unexpected result
Insert a lot of entries into one partition:
```
INSERT INTO default.test_s3_tiered
SELECT
    number AS A,
    substring(toString(rand()), 1, 10) AS S,
    toDate('2023-01-01') AS D
FROM numbers(1000000000);
```
List StorageClass of objects and find out that some of them have STANDARD storage class:
```
aws s3api list-objects-v2 --bucket clickhouse-offload | grep StorageClass | sort | uniq -c
16936             "StorageClass": "INTELLIGENT_TIERING"
 218             "StorageClass": "STANDARD"
```

**Expected behavior**
All objects created in the bucket are expected to have INTELLIGENT_TIERING storage class.

**Additional context**
When single rows are inserted, all objects in s3 are created with INTELLIGENT_TIERING storage class.

When test insert are made to different partitions, with
```
INSERT INTO default.test_s3_tiered
SELECT
    number AS A,
    substring(toString(rand()), 1, 10) AS S,
    toDate('2023-01-01') + (number % 365) AS D
FROM numbers(10000000000);
```
all objects in s3 are created with INTELLIGENT_TIERING storage class.

When partitions are moved from local disks to s3 with `ALTER TABLE default.test_s3_tiered MOVE PARTITION ID '...' TO DISK 's3_'`, some objects s3 are created with STANDARD storage class.

**Comment by den-crane** [comment]
Probably this feature is not used for clikhouse s3 multipart upload

**Comment by 1283386** [comment]
By the way, in 24.7.3.42 all objects are created with STANDARD class, looks like the setting does not work at all there.

**Comment by nickitat** [comment]
in the code i see that `s3_storage_class` is taken into account only for backups. did you find somewhere any mentions of it being supported for normal table storage?

**Comment by den-crane** [comment]
it was intended for s3 disks 
* https://github.com/ClickHouse/ClickHouse/issues/44443
* https://github.com/ClickHouse/ClickHouse/pull/44707

**Comment by den-crane** [comment]
@filimonov

**Comment by den-crane** [comment]
[robot-ch-test-poll4](https://github.com/robot-ch-test-poll4)
closed this as [completed](https://github.com/ClickHouse/ClickHouse/issues?q=is%3Aissue%20state%3Aclosed%20archived%3Afalse%20reason%3Acompleted)[5 hours ago](https://github.com/ClickHouse/ClickHouse/issues/68551#event-26277351869)

I guess it's a bug in robot-ch-test-poll

**Comment by clickgapai** [comment]
_Found via ClickGap automated review. Please close or comment if this is incorrect or needs adjustment._

### TL;DR
**Some objects are created with 'STANDARD' storage class instead of 'INTELLIGENT_TIERING' on s3 disk.**

**ClickGap verified:** reproduced it with the script below on `26.5`; tested 5 version(s) — 4 affected, 1 clean (see Affects); confirmed it predates the oldest available release (pre-existing).

### Reproduction
```
import http.server, socketserver, json, sys, urllib.parse
PORT = int(sys.argv[1])
recorded = {"PutObject": [], "CreateMultipartUpload": []}
H = "x-amz-storage-class"

def classify(cmd, path):
    q = urllib.parse.parse_qs(urllib.parse.urlparse(path).query, keep_blank_values=True)
    if cmd == "POST" and "uploads" in q:
        return "CreateMultipartUpload"
    if cmd == "PUT" and "partNumber" not in q and "uploadId" not in q:
        return "PutObject"
    return None

class Handler(http.server.BaseHTTPRequestHandler):
    def log_message(self, *a): pass
    def _body(self):
        n = self.headers.get("Content-Length")
        if n:
            try: self.rfile.read(int(n))
            except Exception: pass
    def _send(self, code, body=b"", extra=None):
        self.send_response(code)
        if extra:
            for k, v in extra.items(): self.send_header(k, v)
        self.send_header("Content-Length", str(len(body)))
        self.end_headers()
        if body: self.wfile.write(body)
    def do_GET(self):
        if self.path == "/recorded":
            b = json.dumps(recorded).encode()
            self._send(200, b, {"Content-Type": "application/json"}); return
        # ListObjects / object GET: empty, harmless
        self._send(200, b'<?xml version="1.0"?><ListBucketResult></ListBucketResult>')
    def do_HEAD(self):
        self.send_response(404); self.send_header("Content-Length","0"); self.end_headers()
    def do_DELETE(self):
        self._body(); self._send(204)
    def do_PUT(self):
        op = classify("PUT", self.path)
        if op: recorded[op].append(self.headers.get(H))
        self._body()
        self._send(200, b"", {"ETag": '"d41d8cd98f00b204e9800998ecf8427e"'})
    def do_POST(self):
        op = classify("POST", self.path)
        if op: recorded[op].append(self.headers.get(H))
        self._body()
        p = urllib.parse.urlparse(self.path)
        key = p.path
        q = urllib.parse.parse_qs(p.query, keep_blank_values=True)
        if op == "CreateMultipartUpload":
            body = ('<?xml version="1.0" encoding="UTF-8"?>'
                    '<InitiateMultipartUploadResult><Bucket>root</Bucket>'
                    '<Key>%s</Key><UploadId>fakeupload</UploadId>'
                    '</InitiateMultipartUploadResult>' % key).encode()
            self._send(200, body); return
        if "uploadId" in q:  # CompleteMultipartUpload
            body = ('<?xml version="1.0" encoding="UTF-8"?>'
                    '<CompleteMultipartUploadResult><Location>http://x%s</Location>'
                    '<Bucket>root</Bucket><Key>%s</Key>'
                    '<ETag>"d41d8cd98f00b204e9800998ecf8427e"</ETag>'
                    '</CompleteMultipartUploadResult>' % (key, key)).encode()
            self._send(200, body); return
        self._send(200, b"")

class TS(socketserver.ThreadingMixIn, http.server.HTTPServer): pass
TS(("127.0.0.1", PORT), Handler).serve_forever()
```
[▶ Run on ClickHouse Fiddle](https://fiddle.clickhouse.com/4f46742f-3723-4244-bd7d-2635a9a71185)

### Bisect
- **Present since at least `v24.1.8.22`** — reproduces on every release line tested down to there; the introducer predates the oldest release with a downloadable binary, so a single commit isn't pinnable from release binaries (not a recent regression).

### Affects
- ✓ **Fix landed on master** in PR #106214. Branches below still have the bug and need the backport.
- **Reproduces on**: `26.5, 26.4, 26.3, 25.8`
- **Does NOT reproduce on**: `master`
- **Backport needed**: 26.5, 26.4, 26.3, 25.8

### Suggested next step
Backport PR #106214 to the affected release branches listed above. Bisect on `comp-object-storage` already identified the fix; no further triage needed.

**Component:** `comp-object-storage`
**Severity:** `P2`

cc fix PR author: @alexey-milovidov

---
_ClickGap · Finding: `h_issue_68551`_




### Triage metadata
- Primary component: `comp-object-storage`
- Component owners (`comp-object-storage`): @scanhex12 @kssenii
- Related: #78737

**Comment by Felixoid** [comment]
@alexey-milovidov please review the linked "fix" and adjust the claude
